services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_compose_specs.py

```diff
@@ -104,7 +126,92 @@ def _replace_env_vars_in_compose_spec(
     return stringified_service_spec
 
 
-def assemble_spec(
+async def _inject_resource_limits_and_reservations(
+    app: FastAPI, service_key: str, service_tag: str, service_spec: ComposeSpecLabel
+) -> None:
+    director_v0_client: DirectorV0Client = get_director_v0_client(app)
+    labels_for_involved_services: Dict[
+        str, SimcoreServiceLabels
+    ] = await get_labels_for_involved_services(
+        director_v0_client=director_v0_client,
+        service_key=service_key,
+        service_tag=service_tag,
+    )
+
+    # example: '2.3' -> 2 ; '3.7' -> 3
+    docker_compose_major_version: int = int(service_spec["version"].split(".")[0])
+
+    for spec_service_key, spec in service_spec["services"].items():
+        if spec_service_key not in labels_for_involved_services:
+            logger.info(
+                "No labels found for service %s, service may not be an osparc service",
+                spec_service_key,
+            )
+            continue
+        labels = labels_for_involved_services[spec_service_key]
+
+        settings_list: List[SimcoreServiceSettingLabelEntry] = labels.settings
+
+        # defaults
+        limit_nano_cpus = DEFAULT_LIMIT_NANO_CPUS
```

Comment:
It is probably not possible to limit the sum of the resources of all services in the compose-stack? That would make sense for sim4life, since if one of the three services exceeds the limit, the others also wont work anymore. 
Anyway, I wll test whether the default limits work once this goes in.

Reply:
Each service has to be individually limited, because they are started as sperate containers whom happen to be on the same network and host. Only per container limits are enforced. Sharing a limit with other containers is not possible.

I would kindly like you to test it. You can always add some limits and reservations on `rt-web` and `sym-server`. The only one which has it currently is the `core`.

Reply:
UPDATE: by default each service in the stack has its own limits. From the UI, the user will be able to pick and assign resources to each individual service